UI/window-basic-main.cpp

```diff
@@ -2138,6 +2138,82 @@ void OBSBasic::OBSInit()
 
 	/* ----------------------------- */
 	/* add custom browser docks      */
+#if BROWSER_AVAILABLE && YOUTUBE_ENABLED
+#define YOUTUBE_VIDEO_URL "://studio.youtube.com/video/"
+	// remove legacy YouTube Browser Docks (once)
+
+	bool youtube_cleanup_done = config_get_bool(
+		App()->GlobalConfig(), "General", "YtDockCleanupDone");
+
+	if (!youtube_cleanup_done) {
+		config_set_bool(App()->GlobalConfig(), "General",
+				"YtDockCleanupDone", true);
+
+		const char *jsonStr = config_get_string(App()->GlobalConfig(),
+							"BasicWindow",
+							"ExtraBrowserDocks");
+		if (jsonStr) {
+			std::string err;
+			Json json = Json::parse(jsonStr, err);
+			if (!err.empty())
+				return;
+
+			Json::array array = json.array_items();
+
+			QString removedYTUrl;
+			for (Json &item : array) {
+				QString str_url =
+					item["url"].string_value().c_str();
+				if (str_url.contains(YOUTUBE_VIDEO_URL)) {
+					blog(LOG_DEBUG,
+					     "YT: found legacy url: %s",
+					     QT_TO_UTF8(str_url));
+					removedYTUrl += str_url;
+					removedYTUrl += ";\n";
+				}
+			}
+			if (!removedYTUrl.isEmpty()) {
+				Json::array save_array;
+
+				const QString msg_title =
+					QTStr("YouTube.DocksRemoval.Title");
+				const QString msg_text =
+					QTStr("YouTube.DocksRemoval.Text")
+						.arg(removedYTUrl);
+				OBSMessageBox::warning(this, msg_title,
+						       msg_text);
```

Comment:
I personally don't like the fact that we're forcing the removal here. Is there a reason for that? Wouldn't it be better to ask the user whether they want to remove them, with an explanation that there are built-in ones?

Reply:
The reason is that it was possible to get this functionality before using a direct URL (https://studio.youtube. ...), but now this is not a valid way. Moreover, such links will produce internal errors, and will increase questions "why it's not working" and, I think, more error situations on the YT backend.

So, the proper solution is to remove such links, with new dock panel introduction.

Reply:
This is now resolved. Isn't it?

Reply:
> I personally don't like the fact that we're forcing the removal here. Is there a reason for that? Wouldn't it be better to ask the user whether they want to remove them, with an explanation that there are built-in ones?

As mentioned above, the old URLs may fail to work moving forward. In order to ensure the the dock functionality works as expected, they are being replaced as a one-time upgrade operation. We provided the acceptable options of "remove the old docks once ever or not at all" (options approved by @Warchamp7 ) in [an earlier review comment](https://github.com/obsproject/obs-studio/pull/9031#discussion_r1222208327), and the option for "remove old docks once ever" was chosen.

As such, I'll resolve this thread.